proxmoxtf/provider/provider.go

```diff
@@ -209,13 +221,32 @@ func (r *apiResolver) Resolve(ctx context.Context, nodeName string) (ssh.Proxmox
 		}
 	}
 
+	// Fallback 3: All else fails, do a good old DNS lookup
+	// Nodes with only DHCP will fallback here.
+	// Proxmox only provides what "configured" (static),
+	// not what is "assigned" (DHCP)
+	if nodeAddress == "" {
+		tflog.Debug(ctx, "No interfaces with at least a static IPV4 addresss found.")
+		tflog.Debug(ctx, "Attempting a DNS lookup for node.")
+
+		ips, _ := net.LookupIP(nodeName)
```

Comment:
It might be useful to wrap and return the error here, to diagnose any resolver issues.